File: src/test/java/se/alipsa/sourceurl/SourceUrlTest.java
package se.alipsa.sourceurl;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.security.KeyManagementException;
import java.security.KeyStoreException;
import java.security.NoSuchAlgorithmException;
import java.security.UnrecoverableKeyException;
import java.security.cert.CertificateException;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SourceUrlTest {

  @Test
  public void testHttps() throws HttpsException, IOException, NoSuchAlgorithmException, UnrecoverableKeyException, CertificateException, KeyStoreException, URISyntaxException, KeyManagementException {
    int port = TlsServer.start();
    String body = SourceUrl.fetchContent("https://localhost:" + port + "/test", true);
    assertEquals("print('This is the response'); aVar <- 42", body);
    TlsServer.stop();
  }
}

File: src/test/R/SourceurlTest.R
library('hamcrest')
library('se.alipsa:sourceUrl')

test.https <- function() {
  import(se.alipsa.sourceurl.TlsServer)
  port <- TlsServer$start()
  sourceUrl(paste0("https://localhost:", port, "/test"), allowSelfSigned = TRUE)
  assertThat(aVar, equalTo(42))
  TlsServer$stop()
}